ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.ts

```diff
@@ -58,31 +81,130 @@ export class ActivityFileAnswer implements OnDestroy {
                 )
                 .subscribe((res: FileUploadResponse) => {
                     this.isFileSelected = true;
-                    this.fileUploadData = res;
+                    this.fileToUpload = {
+                        ...res,
+                        name: file.name,
+                        size: file.size,
+                        isUploaded: false
+                    };
                 });
         }
     }
 
-    uploadFile(): void {
-        const formData = new FormData();
-        formData.append('file', this.file);
+    submitFileUpload(): void {
+        const failedLocalValidator = this.getFailedLocalValidator();
+        if (failedLocalValidator) {
+            this.errorMessage = failedLocalValidator.result;
+            return;
+        }
+
+        if (this.uploadedFile) {
+            this.openConfirmDialog();
+        } else {
+            this.uploadFile();
+        }
+    }
+
+    public openConfirmDialog(): void {
+        const panelClass = 'file-upload-confirm-dialog';
+        const config = this.modalService.getDialogConfig(this.uploadButtonRef, panelClass);
+        config.data = {
+            title: 'SDK.FileUpload.ConfirmReuploadTitle',
+            confirmBtnText: 'SDK.FileUpload.ReuploadBtnText',
+            cancelBtnText: 'SDK.FileUpload.CancelBtnText'
+        };
+
+        const dialogRef = this.dialog.open(ConfirmDialogComponent, config);
+        dialogRef.afterClosed().subscribe((confirmUpload: boolean) => {
+            if (confirmUpload) {
+                this.uploadFile();
+            }
+        });
+    }
+
+    undoUploadedFile(): void {
+        const panelClass = 'file-upload-confirm-dialog';
```

Comment:
if this panel class is common for all dialogs of this component can we move it to private readonly class property?

Reply:
Fixed at 0f692defcc2f194893faa6d53ca7b38a2ac473fc